src/main/java/dk/kb/storage/facade/DsStorageFacade.java

```diff
@@ -331,7 +377,17 @@ private static void validateIdHasOriginPrefix(String origin, String recordId) {
         }
     }
     
-
+    /**
+     * RecordType is not null
+     * @param origin name.
+     */
+    private static void validateRecordType(RecordTypeDto type) {
+        if (type==null) {
+            throw new InvalidArgumentServiceException("RecordType must not be null");
+        }
+    }
```

Comment:
A question: In theory, if I was interested in ALL records, how would I then get these?

Reply:
The other method on the API (getRecordsModifiedAfter) that does not take recordtype as parameter.
But this will also not expand records to local tree.